mod/consensus/pkg/cometbft/service/abci.go

```diff
@@ -320,9 +316,6 @@ func (s *Service[LoggerT]) internalFinalizeBlock(
 	if s.finalizeBlockState == nil {
 		s.setState(execModeFinalize)
 	}
-	if s.finalizeBlockState == nil {
-		return nil, errors.New("finalizeBlockState is nil")
-	}
```

Comment:
redundant check since `s.setState(execModeFinalize)` is called right before. We don't do these checks after `setState` is called for proposals.